### PR TITLE
feat(timeline): add docs

### PR DESCRIPTION
**Pull request by PixeledCode**

### CODE REVIEW

File: packages/paste-website/src/pages/components/timeline/index.mdx [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+  <ListItem>Hour, Minute, Second (ex: "12:38 pm")</ListItem>
+  <ListItem>Additional Details (ex: "0.2 s" for duration)</ListItem>
+</OrderedList>
+
+<LivePreview scope={{ Timeline, TimelineItem, StatusBadge, Box, Paragraph }} noInline>
+  {TimelineTimestamp}
+</LivePreview>
+
+## When to use Timeline
+
```
Comment (roxanagomez) [inline packages/paste-website/src/pages/components/timeline/index.mdx:286]: **issue (blocking):**  It seems something happened with the examples, the "Event titles" labels are placed on the description, and the 3rd one has only one timelineItem.
Screenshot 2024-09-17 at 3 59 16 PM

File: packages/paste-website/src/pages/components/timeline/index.mdx [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+The events should load in batches of the same number of events that were initially displayed, once all the events have been displayed, remove the “Load more” Button.
+
+### Infinite scroll
+
+Use infinite scroll only when the Timeline is the last item on the page or surface, and there are no footers or other elements below it that the user needs to interact with.
+Use [Skeleton Loader](/components/skeleton-loader) for results that might take a while to load.
+
+<Callout variant="neutral" marginY="space70">
+  <CalloutText>
+    Tip: For very long Timelines, where pressing 'Load more' Buttons might become cumbersome, consider using a{" "}
```
Comment (roxanagomez) [inline packages/paste-website/src/pages/components/timeline/index.mdx:226]: **nitpick:** update 'Load more' to "Load more"

File: packages/paste-website/src/pages/components/timeline/index.mdx [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+        If the value is too verbose for a screen reader or not a clear label for the collapsed content, provide a helpful label for the button using the <inlineCode>collapsibleLabelText</inlineCode> prop on TimelineItem
+      </ListItem>
+    </UnorderedList>
+  </ListItem>
+  <ListItem>
+    Don’t use icons TimelineItem markers to represent status like warning, error, or success. Since the marker is non-interactive, relying on icons for semantic meaning can cause accessibility challenges, particularly for people using screen readers.
+  </ListItem>
+</UnorderedList>
+
+### Timeline vs. Progress Steps
```
Comment (nkrantz) [inline packages/paste-website/src/pages/components/timeline/index.mdx:110]: suggestion: can we add this section to progress steps docs too? in either PR

File: packages/paste-website/src/component-examples/TimelineExamples.ts
Comment (roxanagomez) [inline packages/paste-website/src/component-examples/TimelineExamples.ts:1]: **issue (blocking):**  When a long text is added under the description, the timeline connector becomes misaligned.
Screenshot 2024-09-17 at 4 13 30 PM

File: packages/paste-website/src/pages/components/timeline/index.mdx [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+<LivePreview scope={{ Timeline, TimelineItem, SMSCapableIcon, SentIcon, DisableIcon }} noInline>
+  {TimelineIcon}
+</LivePreview>
+
+### Collapsible Timeline content
+
+Use the `collapsible` prop when your Timeline includes additional information that doesn't need to be visible at all times.
+
+<LivePreview
```
Comment (roxanagomez) [inline packages/paste-website/src/pages/components/timeline/index.mdx:162]: **praise:** this example looks so good 🤩

File: packages/paste-website/src/pages/components/timeline/index.mdx (hunk 3 above)
Comment (PixeledCode) [reply packages/paste-website/src/pages/components/timeline/index.mdx:110]: will do in progress-steps pr

**Review by roxanagomez — APPROVED** [review]
Niceee ✨

**Pull request merged**